### ISSUE TITLE
Matrix distance to scatter plot

### ISSUE THREAD

**Issue by JudKil** [wjm41/molplotly#20]
Hello everyone,

My name is Judith and for my PhD studies, I would like to use your beautiful scripts. 
I get a distance matrix by rmsd between each pose but I don't see how to pass it to a scatter plot of 2 clusters, I tried with pandas but I'm really blocked, I can't select the lines and the columns to generate the scatter plot

Best Regards,
Judith

**Comment by wjm41** [comment]
Hi Judith! Let me try to help :) Could you attach some example code and give more details about what you are trying to plot?

From my understanding, you have some conformers (for the same/different molecules?) and have clustered them into two clusters and want to visualise the poses. What are the axes of the scatter plot you are trying to do? Are they just PCA axes or something like that?

**Comment by JudKil** [comment]
Hello,

I have a portein on which I have docked a ligand with Autodock Vina.

Vina has generated several possible ligand poses (my different conformers). I have drawn a distance matrix between each conformation. I would then like to group them in a cluster via a kmeans.  I succeeded in making the kmeans and I see the scatter plot of my ligands. Yes I want to visualize the positions with the names on the dots. But as I can't add the name. Therefore I don't know which is which on the figure


I send you :
-the script
- the protein in sdf format
- the ligands generated in sdf format.

(KEEEP **you need to change .txt by .sdf**
[lig-dump-rec-1.all_poses.txt](https://github.com/wjm41/molplotly/files/8974396/lig-dump-rec-1.all_poses.txt)
[1.txt](https://github.com/wjm41/molplotly/files/8974406/1.txt)
[pose_clustering_centroid.txt](https://github.com/wjm41/molplotly/files/8974414/pose_clustering_centroid.txt)

**For the script .txt by .ipybd**
 otherwise I couldn't send them to github)
Maybe you would need to make what pip to get the script to run


Thanks,
Judith

**Comment by wjm41** [comment]
The easiest way to add the 'names' of the molecules would be to to add the pose index as a column in the `table` dataframe:

```python
import plotly.express as px

table['pose_index'] = [f'Molecule Index: {index}' for index in table.index]
table['cluster'] = kmeans.labels_
table['cluster'] = table['cluster'].astype(str)

px.scatter(table, 
            x="1", 
            y="2",
            color='cluster',
            hover_name='pose_index',
            width=1000,
            height=800)
```
which would look something like this:

![Screenshot 2022-06-24 at 16 29 38](https://user-images.githubusercontent.com/17111438/175568047-bc1acce4-7a94-4747-a1fd-7b97f63bd0df.png)

I have also modified the molplotly code a bit to show the actual 3D coordinates as well, but I'm not sure how helpful that would be:

![Screenshot 2022-06-24 at 16 30 29](https://user-images.githubusercontent.com/17111438/175568194-1b82e6bf-419b-4d27-8c9b-5060091a3aa1.png)

Let me know if that would be helpful and I can merge it into the package :)

**Comment by JudKil** [comment]
Dear William,

Thank you very much, that's it !  But is there an option to add the centroid of clusters ?

Best !
Judith

**Comment by JudKil** [comment]
Yes, the 3D option is also interesting ! The last but not least, do you know increase the size of the dots, when I try "size", I don't see de arguments format, size=10 or size=10:10:20 don't work 

Best !
Judith

**Comment by wjm41** [comment]
So so sorry for not replying, I had totally missed your follow-up questions! In any case, here are some answers:

> increase the size of the dots

This can be done by adding `scatter_fig.update_traces(marker=dict(size=12))` and adjusting the `size` parameter in the argument!

> is there an option to add the centroid of clusters

This can be done via `scatter_fig.add_trace` - the full example is below:

```python
import plotly.graph_objects as go

rmsd_df['pose_index'] = range(1, len(renamed_poses)+1)
rmsd_df['cluster'] = kmeans.labels_
rmsd_df['cluster'] = rmsd_df['cluster'].astype(str)

scatter_fig = px.scatter(rmsd_df, 
                x="1", 
                y="2",
                color='cluster',
                hover_name='pose_index',
                width=1000,
                height=800,
                title='Clustering of ligand poses',
                labels={'1':'RMSD to pose 1',
                        '2':'RMSD to pose 2',}
                )
scatter_fig.add_trace(
        go.Scatter(
            x = kmeans.cluster_centers_[:,0],
            y = kmeans.cluster_centers_[:,1],
            mode='markers',
            marker=dict(color="red", 
                        symbol='x',
                        size=10),
            showlegend=False,))
scatter_fig.update_traces(marker=dict(size=12))

```
![Screenshot 2022-11-24 at 15 30 18](https://user-images.githubusercontent.com/17111438/203820589-9829cd1b-0798-426d-8508-1144cce2debf.png)

**Comment by wjm41** [comment]
On another note, would you mind if I use your code/data as an example for showing the plotting of 3D coordinates with `molplotly`? :)
